src/models/progress/useProgressQuery.js

```diff
@@ -0,0 +1,28 @@
+import { useRef } from 'react';
+import { useQuery } from 'react-query';
+
+import { getProgressQueryKey } from '../../constants/queryKeys';
+import { isProgressSettled } from '../../utils/progressUtils';
+import {
+  getProgress,
+  queryOptions as defaultQueryOptions,
+} from './progressQueryDefaults';
+
+export default function useProgressQuery(progressGuid, queryOptions) {
+  const isSettledRef = useRef(false);
```

Comment:
is useRef better than useState here? seems like maybe this is just a wilder version of `[isSettled, setIsSettled] = useState(false)`

Reply:
the one difference i can think of without looking at the docs is i dont think updating the ref will trigger a rerender. though in this case i think you'd actually want to rerender ASAP?

Reply:
You are definitely right on this one. I'm going to try to find a native react-query way to update the options, but if I can't I am going to switch to useState.

Reply:
nothing wrong with useState